head_first_design_patterns/tbruinem/json/src/parsed_json.cpp

```diff
@@ -0,0 +1,95 @@
+#include "parsed_json.hpp"
+#include "element.hpp"
+
+#include <assert.h>
+
+namespace json {
+
+ParsedJSON::~ParsedJSON() {
+}
+
+void ParsedJSON::SetName(string name)
+{
+	this->name = name;
+}
+
+void ParsedJSON::SetPtrValue(uintptr_t ptr) {
+	assert(this->value_.pointer == 0);
+	this->value_.pointer = ptr;
+}
+
+void ParsedJSON::SetType(ElementType type)
+{
+	this->type = type;
+}
+
+void ParsedJSON::SetValueNumber(int64_t value)
+{
+	this->value_.bigint = value;
+}
+
+void ParsedJSON::SetValueNULL()
+{
+	this->value_.pointer = 0;
+}
+
+void ParsedJSON::SetValueString(string value)
+{
+	SetPtrValue((uintptr_t)new string(value));
```

Comment:
C++ style reinterpret_cast so your code is uglier and you will feel like changing it so you don't have to do a cast